### ISSUE TITLE
[enhancement] train.py 기능 추가

### ISSUE THREAD

**Issue by hanbyeol1001** [boostcampaitech7/level2-cv-semanticsegmentation-cv-22-lv3#17]
# **추가된 기능에 대한 설명을 작성하세요**
- [x] loss function 선택
- [x] optimizer 선택
- [x] schedular 선택
- [x] loss 출력하는 주기 config로 설정하도록 수정
- [x] val 주기에 맞춰 체크포인트 저장하는 코드 추가
<br><br>

# **수정해야 할 파일들**
- [x] train.py
- [x] trainer.py 
- [x] base_config.yaml
- [x] loss_opt_sche.py
<br><br>

# **추가 전달 사항**
추가적으로 하면 좋을 것들
- Optuna 사용(하이퍼파라미터 튜닝 중에서 도움이 되는 라이브러리)
